components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/vault/utils/Utils.java

```diff
@@ -0,0 +1,300 @@
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+/*
+ *  Contains utils required by the secure vault implementation for the micro integrator profile
+ */
+public class Utils {
+
+    /**
+     * Writes a given alias and encrypted password to the given file path
+     *
+     * @param alias alias of the new user
+     * @param encryptedPass encrypted value of the password given by the user
+     * @param filePath file path where the alias and password needs to be stored
+     */
+    public static void writeKeyToFile(String alias, String encryptedPass, String filePath) {
+        System.out.println("writing to: " + filePath);
+        Properties keys = new Properties();
+        File file = new File(filePath);
+        File fileParent = file.getParentFile();
+        boolean parentExists = fileParent.exists();
+        if (!parentExists) {
+            if (!fileParent.mkdirs()) {
+                String msg = "Error in registry directory structure creation at : " + fileParent.getPath();
+                throw new SecureVaultException(msg);
+            }
+        } else {
+            try (FileInputStream fileInputStream = new FileInputStream(file)) {
+                keys.load(fileInputStream);
+                if (keys.containsKey(alias)) {
+                    String change = System.getProperty(Constants.CONSOLE_PASS_CHANGE_PARAM);
+                    if (change != null && change.equals(Constants.TRUE)) {
+                        keys.remove(alias);
+                        keys.put(alias, encryptedPass);
+                        System.out.println("changed password for alias: " + alias);
+                    } else {
+                        String msg = "An entry with the alias: " + alias + " already exists";
+                        throw new SecureVaultException(msg);
+                    }
+                }
+            } catch (IOException e) {
+                String msg = "Error loading properties from a file at : " + filePath;
+                throw new SecureVaultException(msg + " Error : " + e.getMessage(), e);
+            }
+        }
+        try (FileOutputStream fileOutputStream = new FileOutputStream(file, false)) {
+            keys.setProperty(alias, encryptedPass);
+            //The parent folder is touched to support versioning done by the registry
+            if (parentExists) {
+                touchFile(fileParent);
+            }
+            keys.store(fileOutputStream, "Secure Vault keys for micro integrator");
+        } catch (IOException e) {
+            String msg = "Error loading properties from a file at : " + filePath;
+            throw new SecureVaultException(msg + " Error : " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Update last modified time of a given file to the current time
+     *
+     * @param file file that needs to be modified
+     */
+    public static void touchFile(File file) {
```

Comment:
In micro-integrator we do not support hot deployments. What is the usage is this method?

Reply:
This is used by the file-based registry implementation to create files.